snap/snapcraft.yaml

```diff
@@ -8,39 +8,73 @@ description: |
 grade: stable
 confinement: classic
 
+environment:
+  GEM_HOME: $HOME/.gem
```

Comment:
I do not know about snap well.  When does `$HOME` expand?
If it expands at build time, it makes difference when using `$HOME` in `launch`.
And why remove `$SNAP/lib/ruby/gems/2.5.0` from `GEM_HOME`?

Reply:
So, `GEM_HOME`, yes I did remove it, as it is not a list but a specific 
location to install gems to, while `GEM_PATH` is a list of paths where to find 
gems[1]; the side effect seen here was that gems were being installed into 
`~/.gem:/snap/ruby/...`.

About `HOME`, it is runtime defined, this will do the right thing.

These changes allows one to get further with a call like `gem install 
nokogiri`.

[1] https://stackoverflow.com/questions/11277227/whats-the-difference-between-gem-home-and-gem-path